// flash-msg/bundles/flash.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('flash', ['exports', '@angular/core'], factory) :
    (global = global || self, factory(global.flash = {}, global.ng.core));
}(this, function (exports, core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var FlashService = /** @class */ (function () {
        function FlashService() {
        }
        FlashService.decorators = [
            { type: core.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        FlashService.ctorParameters = function () { return []; };
        /** @nocollapse */ FlashService.ngInjectableDef = core.ɵɵdefineInjectable({ factory: function FlashService_Factory() { return new FlashService(); }, token: FlashService, providedIn: "root" });
        return FlashService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var FlashComponent = /** @class */ (function () {
        function FlashComponent() {
        }
        /**
         * @return {?}
         */
        FlashComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
        };
        FlashComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'lib-flash',
                        template: "\n    <p>\n      flash works!\n    </p>\n  "
                    }] }
        ];
        /** @nocollapse */
        FlashComponent.ctorParameters = function () { return []; };
        return FlashComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var FlashModule = /** @class */ (function () {
        function FlashModule() {
        }
        FlashModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [FlashComponent],
                        imports: [],
                        exports: [FlashComponent]
                    },] }
        ];
        return FlashModule;
    }());

    exports.FlashComponent = FlashComponent;
    exports.FlashModule = FlashModule;
    exports.FlashService = FlashService;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
