// flash-msg/fesm5/flash.js
import { Injectable, ɵɵdefineInjectable, Component, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FlashService = /** @class */ (function () {
    function FlashService() {
    }
    FlashService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    FlashService.ctorParameters = function () { return []; };
    /** @nocollapse */ FlashService.ngInjectableDef = ɵɵdefineInjectable({ factory: function FlashService_Factory() { return new FlashService(); }, token: FlashService, providedIn: "root" });
    return FlashService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FlashComponent = /** @class */ (function () {
    function FlashComponent() {
    }
    /**
     * @return {?}
     */
    FlashComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    FlashComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-flash',
                    template: "\n    <p>\n      flash works!\n    </p>\n  "
                }] }
    ];
    /** @nocollapse */
    FlashComponent.ctorParameters = function () { return []; };
    return FlashComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var FlashModule = /** @class */ (function () {
    function FlashModule() {
    }
    FlashModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [FlashComponent],
                    imports: [],
                    exports: [FlashComponent]
                },] }
    ];
    return FlashModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { FlashComponent, FlashModule, FlashService };
