// flash-msg/fesm2015/flash.js
import { Injectable, ɵɵdefineInjectable, Component, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class FlashService {
    constructor() { }
}
FlashService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
FlashService.ctorParameters = () => [];
/** @nocollapse */ FlashService.ngInjectableDef = ɵɵdefineInjectable({ factory: function FlashService_Factory() { return new FlashService(); }, token: FlashService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class FlashComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
FlashComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-flash',
                template: `
    <p>
      flash works!
    </p>
  `
            }] }
];
/** @nocollapse */
FlashComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class FlashModule {
}
FlashModule.decorators = [
    { type: NgModule, args: [{
                declarations: [FlashComponent],
                imports: [],
                exports: [FlashComponent]
            },] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { FlashComponent, FlashModule, FlashService };
